### PR TITLE
feat: add SessionJWTCreationRequested filter.

### PR DESCRIPTION
**Pull request by cesarsilvaedunext**
## Description

This PR adds a new filter that updates the JWT token's payload of a User by adding extra data when its creation is requested. The filter receives the payload and the User object, and returns the User and the modified payload with any new data included. For example: adds a new field with the permission roles of a certain subproject. Other uses could include: adding the last access datetime of the User, a status, or any other useful information related to the User.

## Installation instructions
- Install course_operations.
- Install openedx-filters including the changes from this PR: change the openedx-filters requirement so it uses this fork and PR: git+https://github.com/Pearson-Advance/openedx-filters.git@pearson-release/olive.main

## Testing instructions
- Call the run_filter method from SessionJWTCreationRequested with a given payload and a user.
- Check that the permission_roles are valid for that user.

### CODE REVIEW

File: openedx_filters/authentication/filters.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+"""
```
Comment (Squirrel18) [inline openedx_filters/authentication/filters.py:1]: If you are willing to use Python type hints, you should use them throughout the file and there are some missing type hints. @cesarsilvaedunext

File: openedx_filters/authentication/filters.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            payload (str): the payload of JWT token to be modified.
+            user (User): Django User related to the JWT token.
+
+        Returns:
+            tuple[dict, User]:
+                - dict: the modified payload of the JWT token.
+                - User: the user of the JWT token.
+        """
+        data = super().run_pipeline(payload=payload, user=user)
+        return data.get("payload"), data.get("user")
```
Comment (Squirrel18) [inline openedx_filters/authentication/filters.py]: Since payload is a dictionary, it's a good practice to return the same object type as the default value. @cesarsilvaedunext

File: openedx_filters/authentication/tests/test_filters.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Tests for ``authentication`` subdomain filters.
+"""
+from unittest.mock import Mock
+
+from django.test import TestCase
+
+from openedx_filters.authentication.filters import SessionJWTCreationRequested
+
+
+class TestSessionJWTCreationRequested(TestCase):
```
Comment (Squirrel18) [inline openedx_filters/authentication/tests/test_filters.py]: You are not using anything from TestCase class.
You are just testing the result, remember that we always test the actual procedure and the call chain. @cesarsilvaedunext

File: openedx_filters/authentication/tests/test_filters.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    """
+
+    def test_session_jwt_creation_requested(self):
+        """
+        Test SessionJWTCreationRequested filter behavior under normal conditions.
+
+        Expected behavior:
+            - The filter should return the payload and the user.
+        """
+        payload = Mock()
```
Comment (Squirrel18) [inline openedx_filters/authentication/tests/test_filters.py]: There is no need to mock payload as it's just a dict object. @cesarsilvaedunext

File: openedx_filters/authentication/filters.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    Trigger:
+        - Repository: openedx/edx-platform
+        - Path: openedx/core/djangoapps/oauth_dispatch/jwt.py
+        - Function or Method: _create_jwt
+    """
+
+    filter_type = "org.openedx.authentication.session.jwt.creation.requested.v1"
+
+    @classmethod
```
Comment (Squirrel18) [inline openedx_filters/authentication/filters.py]: Why is this a classmethod? You are not using cls definition nor changing the way that SessionJWTCreationRequested is instantiated. @cesarsilvaedunext

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: openedx_filters/authentication/filters.py (hunk 1 above)
Comment (cesarsilvaedunext) [reply openedx_filters/authentication/filters.py:1]: Addressed

File: openedx_filters/authentication/filters.py (hunk 5 above)
Comment (cesarsilvaedunext) [reply openedx_filters/authentication/filters.py]: For every PR I see in the openedx-filters repo, the run_filter method is always a classmethod. I believe it's necessary, but perhaps that's a conversation for the upstream PR.

File: openedx_filters/authentication/filters.py (hunk 2 above)
Comment (cesarsilvaedunext) [reply openedx_filters/authentication/filters.py]: Addressed

File: openedx_filters/authentication/tests/test_filters.py (hunk 3 above)
Comment (cesarsilvaedunext) [reply openedx_filters/authentication/tests/test_filters.py]: For the call chain, addressed.
For the TestCase use, same scenario as the classmethod point, in this particular case for the framework to identify the cases as unit tests.

File: openedx_filters/authentication/tests/test_filters.py (hunk 4 above)
Comment (cesarsilvaedunext) [reply openedx_filters/authentication/tests/test_filters.py]: Addressed

File: openedx_filters/authentication/filters.py (hunk 5 above)
Comment (Squirrel18) [reply openedx_filters/authentication/filters.py]: Understood, but for now, if it doesn't add anything valuable to the implementation, we should remove it rather than keep it. Once we open the upstream conversation, we can define whether this makes sense or not. @cesarsilvaedunext

File: openedx_filters/authentication/tests/test_filters.py (hunk 3 above)
Comment (Squirrel18) [reply openedx_filters/authentication/tests/test_filters.py]: We should not add functionalities if we are not going to use them: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it We will not follow bad practices or solutions that do not have a good technical approach, even if it is upstream, we should have our own technical criteria. @cesarsilvaedunext

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: openedx_filters/authentication/tests/test_filters.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+class TestSessionJWTCreationRequested(TestCase):
+    """
+    Test class to verify standard behavior of the filters located in rendering views.
+    You'll find test suites for:
+
+    - SessionJWTCreationRequested
+    """
+
+    def test_session_jwt_creation_requested(self):
```
Comment (Serafin-dev) [inline openedx_filters/authentication/tests/test_filters.py:18]: what do you think of adding of adding a test case that obtains an empty "payload" ? @cesarsilvaedunext 
`
def test_session_jwt_creation_requested_missing_payload(self):
 `

File: openedx_filters/authentication/tests/test_filters.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        """
+        Test SessionJWTCreationRequested filter behavior under normal conditions.
+
+        Expected behavior:
+            - The filter should return the payload and the user.
+        """
+        payload = {'key': 'value'}
```
Comment (Squirrel18) [inline openedx_filters/authentication/tests/test_filters.py:25]: Please, do not mix single and double quotes throughout the file. @cesarsilvaedunext

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

**Review by Squirrel18 — APPROVED** [review]

**Review by Serafin-dev — APPROVED** [review]
